### PR TITLE
Trialpushbranch

### PR DESCRIPTION
**Pull request by s-maddrellmander**
## Changelogs

- _enumerate the changes of that PR._

---

_Checklist:_

- [ ] _Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._
- [ ] _Add tests to cover the fixed bug(s) or the new introduced feature(s) (if appropriate)._
- [ ] _Update the API documentation is a new function is added, or an existing one is deleted._
- [ ] _Write concise and explanatory changelogs above._
- [ ] _If possible, assign one of the following labels to the PR: `feature`, `fix` or `test` (or ask a maintainer to do it for you)._

---

_discussion related to that PR_

### CODE REVIEW

**Comment by s-maddrellmander** [comment]
@kerstink-GC - missed this when demoing, could you do a lint on these changes and add a new PR please! Sorry about that.

File: graphium/nn/pyg_layers/gps_pyg.py [hunk 1]
```diff
@@ -255,17 +255,22 @@ def forward(self, batch: Batch) -> Batch:
...
-        # Apply the residual connection for the node features
+        # Apply the residual connection for the node features and scale the activations by some value to help reduce activation growth
         if self.node_residual:
-            h_local = self.residual_add(h_local, feat_in)
-        # Scale the activations by some value to help reduce activation growth
-        h_local = self.scale_activations(h_local, self.output_scale)
-        # Apply the residual connection for the edge features
+            if self.layer_depth < 1:
+                h_local = self.residual_add(h_local, feat_in)
+                h_local *= 1/self.scale_activations(h_local, self.output_scale)
```
Comment (DomInvivo) [inline graphium/nn/pyg_layers/gps_pyg.py:262]: torch doesn't like in-line operations. I suggest doing
```python
h_local = h_local / self.scale_activations(h_local, self.output_scale)
```

File: graphium/nn/pyg_layers/gps_pyg.py [hunk 2]
```diff
@@ -255,17 +255,22 @@ def forward(self, batch: Batch) -> Batch:
...
         if self.edge_residual and self.use_edges:
-            e_local = self.residual_add(e_local, edges_feat_in)
-        # Scale the activations by some value to help reduce activation growth
-        if self.use_edges:
-            e_local = self.scale_activations(e_local, self.output_scale)
+            if self.layer_depth < 1:
+                e_local = self.residual_add(e_local, edges_feat_in)
+                e_local *= 1/self.scale_activations(e_local, self.output_scale)
+            else:
+                e_local *= 1/self.scale_activations(e_local, self.output_scale)
```
Comment (DomInvivo) [inline graphium/nn/pyg_layers/gps_pyg.py:272]: same everywhere

File: graphium/nn/pyg_layers/gps_pyg.py [hunk 3]
```diff
@@ -255,17 +255,22 @@ def forward(self, batch: Batch) -> Batch:
...
-            h_local = self.residual_add(h_local, feat_in)
-        # Scale the activations by some value to help reduce activation growth
-        h_local = self.scale_activations(h_local, self.output_scale)
-        # Apply the residual connection for the edge features
+            if self.layer_depth < 1:
+                h_local = self.residual_add(h_local, feat_in)
+                h_local *= 1/self.scale_activations(h_local, self.output_scale)
+            else:
+                h_local *= 1/self.scale_activations(h_local, self.output_scale)
+                h_local = self.residual_add(h_local, feat_in)
```
Comment (DomInvivo) [inline graphium/nn/pyg_layers/gps_pyg.py:265]: Why is the order of operations inverted in the `else`? why not drop the residual altogether?

**Pull request merged**